### ISSUE TITLE
misformatting in debugger help message

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1315]
Imported from SourceForge on 2024-07-04 08:56:32
Created by **[slitvinov](https://sourceforge.net/u/slitvinov/)** on 2015-03-29 14:40:13
Original: https://sourceforge.net/p/maxima/bugs/2929

---

After this commit
https://sourceforge.net/p/maxima/code/ci/1db75dd57df153972b36d994ae647708da6f8676

    --- a/src/mdebug.lisp
    +++ b/src/mdebug.lisp
    @@ -405,9 +405,10 @@
           (catch *quit-tag*
     	(unwind-protect
     	     (do () (())
    -	       (format-prompt *debug-io* "~&~@[(~a:~a) ~]"
    -                              (unless (stringp at) "dbm")
    -                              (length *quit-tags*))
    +	       (format-prompt *debug-io* "~a"
    +			      (format nil "~&~@[(~a:~a) ~]"
    +				      (unless (stringp at) "dbm")
    +				      (length *quit-tags*)))
     	       (finish-output *debug-io*)
     	       (setq val
     		     (catch 'macsyma-quit

There is a tiny misformatting in debugger. A newline is missing before (dbm:1).

    :::text
    (%i1) debugmode(true) $
    (%i2) 0/0 $
    expt: undefined: 0 to a negative exponent.
     -- an error.  Entering the Maxima debugger.
    Enter ':h' for help.
    (dbm:1) :help
    Break commands start with ':'. Any unique substring may be used,
    eg :r :re :res all work for :resume.
    
    Command      Description
    -----------  --------------------------------------
     :break      Set a breakpoint in the specified FUNCTION at the
                 specified LINE offset from the beginning of the function.
    
    <SKIP>
    
     :top        Throw to top level
     :_none      Undocumented(dbm:1)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 08:56:33
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2015-04-15 16:15:16
Original: https://sourceforge.net/p/maxima/bugs/2929/#c698

---

- **status**: open --> closed
- **assigned_to**: Leo Butler

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 08:56:37
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2015-04-15 16:15:16
Original: https://sourceforge.net/p/maxima/bugs/2929/#468f

---

Fixed in commit 2f3905f. 

~~~~~~
(%i1) debugmode(true)$

(%i2) 0/0 $

expt: undefined: 0 to a negative exponent.
 -- an error.  Entering the Maxima debugger.
Enter ':h' for help.

(dbm:1) :h
Break commands start with ':'. Any unique substring may be used,
eg :r :re :res all work for :resume.

Command      Description
-----------  --------------------------------------
 :break      Set a breakpoint in the specified FUNCTION at the
             specified LINE offset from the beginning of the function.
             If FUNCTION is given as a string, then it is presumed to be
             a FILE and LINE is the offset from the beginning of the file.
 :bt         Print a backtrace of the stack frames
 :continue   Continue the computation.
 :delete     Delete all breakpoints, or if arguments are supplied delete the
             specified breakpoints
 :disable    Disable the specified breakpoints, or all if none are specified
 :enable     Enable the specified breakpoints, or all if none are specified
 :frame      With an argument print the selected stack frame.
             Otherwise the current frame.
 :help       Print help on a break command or with no arguments on
             all break commands
 :info       Print information about item
 :lisp       Evaluate the lisp form following on the line
 :lisp-quiet Evaluate the lisp form without printing a prompt
 :next       Like :step, except that subroutine calls are stepped over
 :quit       Quit this level
 :resume     Continue the computation.
 :step       Step program until it reaches a new source line
 :top        Throw to top level
 :_none      Undocumented
(dbm:1) :top

(%i3) 
~~~~~~
